### PR TITLE
Fix #782: add support for `http-proxy` option of podman

### LINKED ISSUE #782: add support for `http-proxy` option
**Is your feature request related to a problem? Please describe.**
By default proxy environment variables are passed into the container if set for the Podman process. This can be disabled by setting the `http-proxy` value to **false**(defaults to **true**) with `--http-proxy=false`. Currently podman-compose does not support this feature.

**Describe the solution you'd like**
If the yaml file contains `http-proxy: false`, then add  `--http-proxy=false` to args.

### PR DESCRIPTION
**Pull request by ChuJiani**
#782

### CODE REVIEW

**Review by JAORMX — APPROVED** [review]

**Review by NerdToMars — APPROVED** [review]

**Review by p12tic — CHANGES_REQUESTED** [review]
Thanks for the PR.

Please add unit tests. See e.g. pytests/test_container_to_args.py for an example.

**Comment by ChuJiani** [comment]
Unit test added in `pytests/test_container_to_args.py`.

Are there any need to place this option into or behind [net_args](https://github.com/containers/podman-compose/blob/121b5f6ea1b6bc1f8ea4ddf4ebc7ff809d5b876e/podman_compose.py#L926)?

**Comment by p12tic** [comment]
> Also: Do I need to place this option into or behind net args

There's stuff like `--dns` ten lines below your change, so I think it doesn't matter for now.

The PR looks good. Could you rebase on top of `main` instead of merging? This whole PR can be a single commit.

File: podman_compose.py [hunk 1]
```diff
@@ -889,6 +889,8 @@ async def container_to_args(compose, cnt, detached=True):
         podman_args.extend(["--annotation", a])
     if cnt.get("read_only", None):
         podman_args.append("--read-only")
+    if cnt.get("http_proxy", None) == False:
```
Comment (p12tic) [inline podman_compose.py:892]: This should be `is False`.

**Comment by ChuJiani** [comment]
Sure! I will figure it out.

**Comment by ChuJiani** [comment]
Sorry I messed things up. It took me some time to rebase/reset. But It should work fine now.

By the way, it seems that the commit just before mine, which is about `pid`, does not pass the unit test.

**Comment by p12tic** [comment]
Thanks, looks great!

**Pull request merged**